datatypes/src/raster/raster_tile.rs

```diff
@@ -1,96 +1,188 @@
-use super::{BaseRaster, Dim2D, Dim3D, GeoTransform, GridDimension, Raster};
+use super::{
+    ArrayShape2D, ArrayShape3D, GeoTransform, GridArray, GridIdx, GridIdx2D, GridIndexAccess,
+    GridIndexAccessMut, GridSize, GridSpaceToLinearSpace, Raster,
+};
 use crate::primitives::{BoundingBox2D, SpatialBounded, TemporalBounded, TimeInterval};
 use crate::raster::data_type::FromPrimitive;
 use crate::raster::Pixel;
+use crate::util::Result;
 use num_traits::AsPrimitive;
 use serde::{Deserialize, Serialize};
 
-pub type RasterTile2D<T> = RasterTile<Dim2D, T>;
-pub type RasterTile3D<T> = RasterTile<Dim3D, T>;
+pub type RasterTile2D<T> = RasterTile<ArrayShape2D, T>;
+pub type RasterTile3D<T> = RasterTile<ArrayShape3D, T>;
 
 /// A `RasterTile2D` is the main type used to iterate over tiles of 2D raster data
 #[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
 pub struct RasterTile<D, T>
 where
+    D: GridSize + GridSpaceToLinearSpace,
     T: Pixel,
 {
     pub time: TimeInterval,
-    pub tile: TileInformation,
-    pub data: BaseRaster<D, T, Vec<T>>,
+    pub tile_position: GridIdx2D,
```

Comment:
some documentation about the tile_position would be nice. maybe even wrap the GridIdx with a newtype to not confuse them with GridArray indexes

Reply:
done